CenterForOpenScience/osf.io
# Add more metrics to nightly keen runs [OSF-8505]

## Purpose
There are a lot of metrics we don't store, including stickiness, and other daily change values. Many of which are calculated using multiple queries, etc.


## Changes
- Add a boat-load of daily queries
- Add stickiness as calculated from keen in one query to use in the future
- Add embargo_v2 that uses future embargo dates instead of simply private registrations
- Swap the variable metrics used for embargoes to the proper value since they are currently broken (institutional embargo counts)
- Update "--yesterday" flag to use timezone.now for all summary files
- Update base keen to use timezone.now so that tests don't fail between 8 and midnight
- Add tests to users and nodes (institutions do not have tests yet)
- Consolidate tests to speed up performance (despite more tests, users 42 seconds to 14; nodes 123 seconds to 28)



## Side effects
None known



## Ticket

https://openscience.atlassian.net/browse/OSF-8505


## QA Considerations
Check that institution metrics show up.
Do a quick pass of metrics on admin and make sure they look like they are updating properly: note any odd data, broken charts, etc.
